pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -713,22 +788,25 @@ func (r *ReconcileSyncSetInstance) updateSyncSetInstanceStatus(ssi *hivev1.SyncS
 }
 
 func (r *ReconcileSyncSetInstance) setUnknownObjectSyncCondition(syncSetConditions []hivev1.SyncCondition, err error, index int) []hivev1.SyncCondition {
-	status := corev1.ConditionFalse
-	reason := unknownObjectNotFoundReason
-	message := fmt.Sprintf("Info available for all SyncSet resources")
-	if err != nil {
-		status = corev1.ConditionTrue
-		reason = unknownObjectFoundReason
-		message = fmt.Sprintf("Unable to gather Info for SyncSet resource at index %v in resources: %v", index, err)
-	}
-	syncSetConditions = controllerutils.SetSyncCondition(
+	return controllerutils.SetSyncCondition(
 		syncSetConditions,
 		hivev1.UnknownObjectSyncCondition,
-		status,
-		reason,
-		message,
-		controllerutils.UpdateConditionNever)
-	return syncSetConditions
+		corev1.ConditionTrue,
+		unknownObjectFoundReason,
+		fmt.Sprintf("Unable to gather Info for SyncSet resource at index %v in resources: %v", index, err),
+		controllerutils.UpdateConditionNever,
```

Comment:
Why is the `UpdateConditionNever`? I would think that you would want it to be updated if the resource with the problem changed. For example, if there was a problem with index 1 and 3 and the user resolved the issue with index 1, you would want the condition to then show that there was a problem with index 3 instead of continuing to show that there was a problem with index 1.

Reply:
I've changed this to `UpdateConditionIfReasonOrMessageChange` so that we get an updated condition message in that case.